web/contribute.rst

```diff
@@ -1,230 +1,458 @@
+
+    conda create -n skbio-dev -c conda-forge --file ci/conda_requirements.txt --file ci/requirements.test.txt --file ci/requirements.lint.txt --file ci/requirements.doc.txt
+
+When done, activate the environment::
 
     conda activate skbio-dev
 
-.. note:: This may be slightly different depending on the operating system. Refer to the Conda site to find instructions for your OS.
+.. note:: This may be slightly different depending on the operating system. Refer to the `Conda documentation <https://docs.conda.io/>`_ to find instructions for your OS.
 
-5. Install scikit-bio::
+4. Install scikit-bio from source code::
 
     pip install --no-deps -e .
 
-6. Test the installation::
+This will install scikit-bio to the current conda environment. After this, you can use scikit-bio like a normal user (e.g., you can do ``import skbio`` in Python code). When you edit the code in the this directory, the changes will be immediately reflected as you use the software.
+
+5. Test the installation::
 
     make test
 
+This will run all unit tests implemented in the scikit-bio codebase to check if the corresponding functionality works correctly. The output should only indicate passes and warnings, but no failures.
 
-Coding guidelines
------------------
+6. Activate pre-commit hooks::
 
-We adhere to the `PEP 8 <https://peps.python.org/pep-0008/>`_ Python style guidelines. Please see scikit-bio's :doc:`coding guidelines <devdoc/code_guide>` for more details. Before submitting code to scikit-bio, you should read this document carefully and apply the guidelines in your code.
+    pre-commit install
 
+This will enable a set of tools that will automatically execute every time you commit changes to ensure code quality.
 
-Testing guidelines
-------------------
 
-All code that is added to scikit-bio must be unit tested, and the unit test code must be submitted in the same pull request as the library code that you are submitting. We will only merge code that is unit tested and that passes the :repo:`continuous integration build <blob/main/.github/workflows/ci.yml>`. This build includes, but is not limited to, the following checks:
+Write code
+----------
 
-- Full unit test suite and doctests execute without errors in supported versions of Python 3.
-- C code can be correctly compiled.
-- Cython code is correctly generated.
-- All tests import functionality from the appropriate minimally deep API.
-- Documentation can be built.
-- Current code coverage is maintained or improved.
-- Code passes ``flake8`` checks.
+Before you start writing code, you may discuss with the scikit-bio team to make sure that your intended contribution is relevant (see `Before coding`_ above). Next, you may work through the following steps to start coding.
+
+1. Update your main branch such that it has the latest version of all files. This is especially important if you cloned a long time ago::
+
+    git checkout main
+    git pull upstream main
```

Comment:
It may be good to mention doing a `git push origin main` at this step. It's not necessary but I believe it is a good practice to keep your forked repo's `main` branch up to date as well.